prometheus/vec.go

```diff
@@ -193,10 +186,29 @@ func (m *MetricVec) Delete(labels Labels) bool {
 	if err != nil {
 		return false
 	}
-	if _, ok := m.children[h]; !ok {
+
+	return m.deleteByHash(h, labels)
+}
+
+// deleteByHash removes the metric from the hash bucket h. If there are
+// multiple matches in the bucket, use lvs to select a metric and remove only
+// that metric.
+func (m *MetricVec) deleteByHash(h uint64, values interface{}) bool {
```

Comment:
Yes, avoiding allocations in this code path is essential. It is very likely to be called very frequently by users.

Current code is the best solution, I guess.